packages/muelu/src/Misc/MueLu_InterfaceAggregationFactory_decl.hpp

```diff
@@ -56,6 +56,51 @@ namespace MueLu
 /*!
     @class InterfaceAggregationFactory class.
     @brief Factory for building aggregates for Lagrange multipliers.
+
+    ## Context, assumptions, and use cases ##
+
+    This factory is intended to be used for saddle-point systems of surface-coupled problems,
+    where constraints are enforced with Lagrange multipliers.
+    In addition to the primal unknowns, Lagrange multipliers are considered as dual unknowns.
+    The presenence of Lagrange mulitpliers make this a prima/dual problem
+
+    It is assumed that each primal slave-side interface node (carrying primal unknowns) is replicated
+    with a dual node carrying the dual unknowns.
+    Furthermore, the number of degrees of freedom per dual node is required to constant for all dual nodes.
+
+    ## Idea ##
+
+    This factory will generate aggregates for the dual nodes such that they "match" the aggregates of their primal counterpart.
+    Instead of performing an actual aggregation procedure on the dual nodes,
+    we grep the existing primal aggregates and use a user-given mapping of dual-to-primal node IDs
+    to create the dual aggregates.
+
+    @ingroup Aggregation
+
+    ## Input/output of InterfaceAggregationFactory ##
+
+    ### User parameters of InterfaceAggregationFactory ###
+    Parameter | type | default | master.xml | validated | requested | description
+    ----------|------|---------|:----------:|:---------:|:---------:|------------
+    A                       | Factory | null |   | * | * | Generating factory of the matrix A
+    Aggregates              | Factory | null |   | * | * | Generating factory of the aggregates (of type "Aggregates" produced, e.g., by the UncoupledAggregationFactory)
+    DualNodeID2PrimalNodeID | Factory | null |   | * | * | Generating factory of the fine dual-to-primal node mapping
+
+    The * in the @c master.xml column denotes that the parameter is defined in the @c master.xml file.<br>
+    The * in the @c validated column means that the parameter is declared in the list of valid input parameters (see InterfaceAggregationFactory::GetValidParameters).<br>
+    The * in the @c requested column states that the data is requested as input with all dependencies (see InterfaceAggregationFactory::DeclareInput).
+
+    The mapping of dual-to-primal node IDs, \c DualNodeID2PrimalNodeID, is of data type \c std::map<LocalOrdinal,LocalOrdinal>.
+    The 'key' refers to the global ID of the dual node, while the 'value' represents the global ID of its primal counterpart.
```

Comment:
The IDs are actually stored as local ID and not global ID.
Therefore, the line:

> The 'key' refers to the global ID of the dual node, while the 'value' represents the global ID of its primal counterpart.

Should be replaced by:

The 'key' refers to the local ID of the dual node, while the 'value' represents the local ID of its primal counterpart.

We can argue on the use of a `std::map` and consider a `std::vector`. However, Moertel (the Trilinos package used to compute the Mortar matrices) relies on a map currently.

Reply:
I'm fine with `std::map` if this enhances compatibility wit `Moertel`.